### PR TITLE
[ticket/16189] Deprecate inet_ntop and inet_pton wrappers

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-16189

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16189

### CODE REVIEW

File: phpBB/includes/functions.php [hunk 1]
```diff
@@ -2200,7 +2200,7 @@ function confirm_box($check, $title = '', $hidden = '', $html_body = 'confirm_bo
 		return false;
 	}
 
-	// re-add sid / transform & to &amp; for user->page (user->page is always using &)
+	// re-add sid / transform & to &amp; for user->page (uhttps://github.com/rubencm/phpbb/pull/new/ticket/15958ser->page is always using &)
```
Comment (3D-I) [inline phpBB/includes/functions.php]: errors in the url, should be:
`// re-add sid / transform & to &amp; for user->page (https://github.com/rubencm/phpbb/pull/new/ticket/15958 user->page is always using &)`

**Comment by marc1706** [comment]
Do the now deprecated functions behave differently than their replacement? If not, shouldn't the functions in functions_compatibility.php just be wrappers to the now used functions?

File: phpBB/includes/functions.php [hunk 2]
```diff
@@ -2851,10 +2851,13 @@ function get_preg_expression($mode)
 		// Whoa these look impressive!
 		// The code to generate the following two regular expressions which match valid IPv4/IPv6 addresses
 		// can be found in the develop directory
+
+		// @deprecated
```
Comment (marc1706) [inline phpBB/includes/functions.php]: I think there isn't that much of a point to deprecate these. The preg expressions are still valid and might also be used by extensions.

Comment (rubencm) [reply phpBB/includes/functions.php]: It have no sense to keep that when php already do that natively with filter_var

**Comment by marc1706** [comment]
Can you rebase onto latest 3.3.x? Thanks.

**Comment by rubencm** [comment]
Done

**Pull request merged**